scipy/optimize/_linprog_util.py

```diff
@@ -368,84 +374,60 @@ def _clean_inputs(lp):
+
+    # Determine shape of provided bounds
+    # np.array(..,dtype=float) raises an error if dimensions are inconsistent
+    # or if there are invalid data types in bounds. Add a linprog header to
+    # the error returned by np.array().
     try:
-        if isinstance(bounds, str):
-            raise TypeError
-        if bounds is None or len(bounds) == 0:
-            bounds = [(0, None)] * n_x
-        elif len(bounds) == 1:
-            b = bounds[0]
-            if len(b) != 2:
-                raise ValueError(
-                    "Invalid input for linprog: exactly one lower bound and "
-                    "one upper bound must be specified for each element of x")
-            bounds = [b] * n_x
-        elif len(bounds) == n_x:
-            try:
-                len(bounds[0])
-            except BaseException:
-                bounds = [(bounds[0], bounds[1])] * n_x
-            for i, b in enumerate(bounds):
-                if len(b) != 2:
-                    raise ValueError(
-                        "Invalid input for linprog, bound " +
-                        str(i) +
-                        " " +
-                        str(b) +
-                        ": exactly one lower bound and one upper bound must "
-                        "be specified for each element of x")
-        elif (len(bounds) == 2 and np.isreal(bounds[0])
-                and np.isreal(bounds[1])):
-            bounds = [(bounds[0], bounds[1])] * n_x
-        else:
-            raise ValueError(
-                "Invalid input for linprog: exactly one lower bound and one "
-                "upper bound must be specified for each element of x")
-
-        clean_bounds = []  # also creates a copy so user's object isn't changed
-        for i, b in enumerate(bounds):
-            if b[0] is not None and b[1] is not None and b[0] > b[1]:
-                raise ValueError(
-                    "Invalid input for linprog, bound " +
-                    str(i) +
-                    " " +
-                    str(b) +
-                    ": a lower bound must be less than or equal to the "
-                    "corresponding upper bound")
-            if b[0] == np.inf:
-                raise ValueError(
-                    "Invalid input for linprog, bound " +
-                    str(i) +
-                    " " +
-                    str(b) +
-                    ": infinity is not a valid lower bound")
-            if b[1] == -np.inf:
-                raise ValueError(
-                    "Invalid input for linprog, bound " +
-                    str(i) +
-                    " " +
-                    str(b) +
-                    ": negative infinity is not a valid upper bound")
-            lb = float(b[0]) if b[0] is not None and b[0] != -np.inf else None
-            ub = float(b[1]) if b[1] is not None and b[1] != np.inf else None
-            clean_bounds.append((lb, ub))
-        bounds = clean_bounds
+        b = np.array(bounds, dtype=float)
```

Comment:
Can this just be `bounds = np.array(bounds, dtype=float)`? Do we need the original bounds any more? `b` suggests right hand side vector.

Reply:
You are right, `b` is not a proper choice for a variable name here. Reusing `bounds` is possible, but would hide the change. I was trying to avoid a long variable name since it is only used in the 10 lines to follow.